apps/infra-local/scripts/stack-up.sh

```
@@ -0,0 +1,238 @@
+[ -L "${APPS_DIR}/apps" ] || ln -sf . "${APPS_DIR}/apps"
+
+# ---------- Component starters ----------
+start_api() {
+  if pid=$(component_pid api); [ -n "$pid" ]; then
+    ok "api already running (PID $pid)"
+    return
+  fi
+  # Defense against stale listeners from a crashed prior session.
+  if port_listening "${PORT_API}"; then
+    warn "port ${PORT_API} already in use — killing prior listener"
+    lsof -ti :${PORT_API} -sTCP:LISTEN | xargs -r kill -9 2>/dev/null || true
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)